tests/test_encoding/test_similarity_encoder.py

```diff
@@ -213,3 +213,162 @@ def test_get_feature_names_out_na(df_enc_big_na):
     }
     assert tr.get_feature_names_out(input_features=None) == out
     assert tr.get_feature_names_out(input_features=input_features) == out
+
+
+def test_keywords_bad_type():
```

Comment:
could we test more than just a string by using parametrize? maybe tuples, integers?